evm/src/generation/mod.rs

```diff
@@ -149,6 +149,17 @@ fn apply_metadata_and_tries_memops<F: RichField + Extendable<D>, const D: usize>
             GlobalMetadata::ReceiptTrieRootDigestAfter,
             h2u(trie_roots_after.receipts_root),
         ),
+        (
+            GlobalMetadata::KernelHash,
+            KERNEL
+                .code_hash
+                .iter()
+                .enumerate()
+                .fold(0.into(), |acc, (i, &elt)| {
+                    acc + (U256::from(elt) << (224 - 32 * i))
+                }),
```

Comment:
Why not something like `U256::from_big_endian(&KERNEL.code_hash)`?

Reply:
`KERNEL.code_hash` is an array of `u32`, do you have an easier way to do it?

Reply:
IIRC, this was stored in limbs only because of KeccakSpongeStark. Now that we do the check inside the Kernel, we can change `code_hash` type to be the actual keccak output (i.e. `[u8; 32]`) and not have issues with the conversion above anymore.